src/components/user-view/raw-text-document/huge-rte/index.tsx

```diff
@@ -0,0 +1,212 @@
+} from 'react';
+import { Editor } from '@hugerte/hugerte-react';
+import { ColumnDiv } from '../../../../styled-components';
+import {
+  CircularProgress,
+  TextField,
+  IconButton,
+  Typography,
+  Box,
+} from '@mui/material';
+import { DocData, DocService } from '../../../../types';
+import { useWithRawTextDocVersions } from '../../../../hooks/use-with-raw-text-doc-versions';
+import { getDocData } from '../../../../hooks/api';
+import debounce from 'lodash/debounce';
+import { useWithUsersDocs } from '../../../../exported-files';
+import EditIcon from '@mui/icons-material/Edit';
+import SaveIcon from '@mui/icons-material/Save';
+import CancelIcon from '@mui/icons-material/Cancel';
+import './huge-rte.css';
+
+interface RawTextDocumentProps {
+  docId?: string;
+  currentActivityId: string;
+}
+
+export function HugeRTEEditor({
+  docId,
+  currentActivityId,
+}: RawTextDocumentProps) {
+  const [docData, setDocData] = useState<DocData | undefined>();
+  const [initialDocData, setInitialDocData] = useState<DocData | undefined>();
+  const [loading, setLoading] = useState<boolean>(!!docId);
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [tempTitle, setTempTitle] = useState('');
+  useWithRawTextDocVersions(currentActivityId, docData);
+  const { updateDocTitleLocally } = useWithUsersDocs();
+
+  const editor = useRef<Editor>(null);
+
+  const editorConfig = useMemo(
+    () => ({
+      plugins: ['markdown', 'lists'],
+      toolbar: [
+        { name: 'history', items: ['undo', 'redo'] },
+        { name: 'styles', items: ['styles'] },
+        { name: 'formatting', items: ['bold', 'italic'] },
+        {
+          name: 'alignment',
+          items: ['alignleft', 'aligncenter', 'alignright', 'alignjustify'],
+        },
+        { name: 'indentation', items: ['outdent', 'indent'] },
+        { name: 'lists', items: ['unordered', 'ordered'] },
+      ],
+    }),
+    []
+  );
+
+  // Debounced update function
+  const debouncedUpdate = useMemo(
+    () =>
+      debounce((markdownText: string) => {
+        const rawText = editor.current?.editor?.getContent({ format: 'text' });
+        if (docData) {
+          setDocData({
+            ...docData,
+            plainText: rawText || markdownText,
+            markdownText: markdownText,
+            lastChangedId: docData.lastChangedId || '',
+          });
+        }
+      }, 500),
+    [docData]
+  );
+
+  useEffect(() => {
+    if (docId) {
+      setLoading(true);
+      getDocData(docId, DocService.RAW_TEXT)
+        .then((docData) => {
+          setDocData(docData);
+          setInitialDocData(docData);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  }, [docId]);
+
+  // Cleanup debounce on unmount
+  useEffect(() => {
+    return () => {
+      debouncedUpdate.cancel();
+    };
+  }, [debouncedUpdate]);
+
+  const handleEditorChange = useCallback(
+    (value: string) => {
+      debouncedUpdate(value);
```

Comment:
`debouncedUpdate` is a `debounce` function, and it's already wrapped in a `useMemo` hook. I don't think there's need to wrap it in a `useCallback`. You could call it directly.